src/Build.UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -1138,6 +1137,85 @@ public void HasMetadata()
             logger.AssertLogContains("[One|Three|Four]");
         }
 
+        [Fact]
+        public void DirectItemMetadataReferenceShouldBeCaseInsensitive()
+        {
+            MockLogger logger = Helpers.BuildProjectWithNewOMExpectSuccess(@"
+                <Project>
+                  <ItemGroup>
+                    <Foo Include=`Foo`>
+                      <SENSITIVE>X</SENSITIVE>
+                    </Foo>
+                  </ItemGroup>
+                  <Target Name=`Build`>
+                    <Message Importance=`high` Text=`QualifiedNotMatchCase %(Foo.FileName)=%(Foo.sensitive)`/>
+                    <Message Importance=`high` Text=`QualifiedMatchCase %(Foo.FileName)=%(Foo.SENSITIVE)`/>
+                    
+                    <Message Importance=`high` Text=`UnqualifiedNotMatchCase %(Foo.FileName)=%(sensitive)`/>
+                    <Message Importance=`high` Text=`UnqualifiedMatchCase %(Foo.FileName)=%(SENSITIVE)`/>
+                  </Target>
+                </Project>
+                ");
+
+            logger.AssertLogContains("QualifiedNotMatchCase Foo=X");
+            logger.AssertLogContains("QualifiedMatchCase Foo=X");
+            logger.AssertLogContains("UnqualifiedNotMatchCase Foo=X");
+            logger.AssertLogContains("UnqualifiedMatchCase Foo=X");
```

Comment:
I was trying to avoid using the log assertions, because i thought they would introduce lots of overhead and asserting via lookup over large strings didn't feel as precise to me? Is there big advantage of these i am missing?

Reply:
IMO the advantage is consistency with the other similar tests, which outweighs the efficiency problems.

Reply:
That's true. I can see that.